void does not have this logic, so the message is useless

--- a/init_functions
+++ b/init_functions
@@ -350,17 +350,6 @@
         elif bitfield_has_bit "$fsckret" 128; then
             err "fatal error invoking fsck"
         fi
-
-        # ensure that root is going to be mounted rw. Otherwise, systemd
-        # might fsck the device again. Annoy the user so that they fix this.
-        if [ "${rwopt:-ro}" != 'rw' ]; then
-            echo "********************** WARNING **********************"
-            echo "*                                                   *"
-            echo "*  The root device is not configured to be mounted  *"
-            echo "*  read-write! It may be fsck'd again later.        *"
-            echo "*                                                   *"
-            echo "*****************************************************"
-        fi
     fi
 }
 
